tools/c7n_azure/c7n_azure/provider.py

```diff
@@ -29,8 +35,18 @@ class Azure(Provider):
     resource_prefix = 'azure'
     resources = PluginRegistry('%s.resources' % resource_prefix)
     resource_map = ResourceMap
+    region_to_cloud = {
+        'AzureCloud': AZURE_PUBLIC_CLOUD,
+        'AzureChinaCloud': AZURE_CHINA_CLOUD,
+        'AzureGermanyCloud': AZURE_GERMAN_CLOUD,
+        'AzureUSGovernment': AZURE_US_GOV_CLOUD
+    }
+
+    cloud = None
 
     def initialize(self, options):
+        self.cloud = self._get_cloud(options)
+
```

Comment:
can we rename this to endpoint_url or cloud_endpoint_url  .. cloud feels a bit nebulous

please also explicitly set options['region'] to the so it can be used as a cache key per #5447